Additional type docs (#816)
hyperledger/firefly: docs/reference/types/_includes/group_description.md

> specified in-line, if it is omitted.

### Group identity hash

The identifying hash for a group is determined as follows:

- All identities are resolved to their DID.
  - An organization name or identity UUID can be used on input
- The UUID of the node that should receive the data for each participant is
  determined (if not specified).
  - The first node found that is in the same identity hierarchy as the
    participant identity, will be chosen.
- The list of participants is ordered by DID, with de-duplication of
  any identities.
- The `namespace`, `name`, and `members` array are then serialized into
  a JSON object, without whitespace.
- A SHA256 hash of the JSON object is calculated

### Private messaging architecture

The mechanism that keeps data private and ordered, without leaking data to the
blockchain, is summarized in the below diagram.

The key points are:

- Data is sent off-chain to all participants via the Data Exchange plugin
  - The Data Exchange is responsible for encryption and off-chain identity verification
- Only parties that are involved in the privacy group receive the data
  - Other parties are only able to view the blockchain transaction

> - The hash and member list of the group not shared outside of the privacy group

Reviewer: ```suggestion
- The hash and member list of the group are not shared outside of the privacy group
```

>   - The `name` of the group can be used to additional salt in generation of the group hash

Reviewer: ```suggestion
  - The `name` of the group can be used as an additional salt in generation of the group hash
```
?

>   - The member list must be known by all members to verify the blockchain transactions,

Reviewer: ```suggestion
  - The member list must be known by all members of the group to verify the blockchain transactions,
```
As opposed to all members of the network
